diff --git a/src/kded/audioshortcutsservice.cpp b/src/kded/audioshortcutsservice.cpp
index a070c78..9ab1a5a 100644
--- a/src/kded/audioshortcutsservice.cpp
+++ b/src/kded/audioshortcutsservice.cpp
@@ -237,7 +237,8 @@ int AudioShortcutsService::changeVolumePercent(PulseAudioQt::Device *device, int
 {
     const qint64 oldVolume = device->volume();
     const int oldPercent = volumePercent(oldVolume);
-    const int targetPercent = oldPercent + deltaPercent;
+    const int step = std::abs(deltaPercent);
+    const int targetPercent = (deltaPercent > 0) ? (oldPercent / step + 1) * step : ((oldPercent - 1) / step) * step;
     const int maxVolume = PulseAudioQt::normalVolume() * (m_globalConfig->raiseMaximumVolume() ? 150 : 100) / 100.0;
     const qint64 newVolume = boundVolume(std::round(PulseAudioQt::normalVolume() * (targetPercent / 100.f)), maxVolume);
     const int newPercent = volumePercent(newVolume);
