diff --git a/nix_update/update.py b/nix_update/update.py
index ca4038c..741d2b5 100644
--- a/nix_update/update.py
+++ b/nix_update/update.py
@@ -169,13 +169,6 @@ def run_update_script(package: Package, opts: Options) -> None:
 
     run(
         [
-            "nix",
-            "develop",
-            *opts.extra_flags,
-            "--impure",
-            "--expr",
-            f"with import <nixpkgs> {{}}; pkgs.mkShell {{inputsFrom = [{opts.get_package()}];}}",
-            "--command",
             "bash",
             "-c",
             " ".join(
